src/components/webViewRe.re

```diff
@@ -0,0 +1,289 @@
+    onLayout::(RNEvent.NativeLayoutEvent.t => unit)? =>
+    onMagicTap::(unit => unit)? =>
+    responderHandlers::Props.touchResponderHandlers? =>
+    pointerEvents::[ | `auto | `none | `boxNone | `boxOnly]? =>
+    removeClippedSubviews::bool? =>
+    style::StyleRe.t? =>
+    testID::string? =>
+    accessibilityComponentType::
+      [ | `none | `button | `radiobutton_checked | `radiobutton_unchecked]? =>
+    accessibilityLiveRegion::[ | `none | `polite | `assertive]? =>
+    collapsable::bool? =>
+    importantForAccessibility::[ | `auto | `yes | `no | `noHideDescendants]? =>
+    needsOffscreenAlphaCompositing::bool? =>
+    renderToHardwareTextureAndroid::bool? =>
+    accessibilityTraits::
+      list [
+        | `none
+        | `button
+        | `link
+        | `header
+        | `search
+        | `image
+        | `selected
+        | `plays
+        | `key
+        | `text
+        | `summary
+        | `disabled
+        | `frequentUpdates
+        | `startsMedia
+        | `adjustable
+        | `allowsDirectInteraction
+        | `pageTurn
+      ]? =>
+    accessibilityViewIsModal::bool? =>
+    shouldRasterizeIOS::bool? =>
+    automaticallyAdjustContentInsets::bool? =>
+    contentInsets::webViewContentInsets? =>
+    injectJavaScript::(string => unit)? =>
+    injectedJavaScript::string? =>
+    mediaPlaybackRequiresUserAction::bool? =>
+    onError::(webViewNavState => unit)? =>
+    onLoad::(webViewNavState => unit)? =>
+    onLoadEnd::(webViewNavState => unit)? =>
+    onLoadStart::(webViewNavState => unit)? =>
+    onMessage::(RNEvent.NativeEvent.t => unit)? =>
+    onNavigationStateChange::(webViewNavState => unit)? =>
+    renderError::(unit => ReasonReact.reactElement)? =>
+    renderLoading::(unit => ReasonReact.reactElement)? =>
+    scalesPageToFit::bool? =>
+    source::webViewSource? =>
+    startInLoadingState::bool? =>
+    domStorageEnabled::bool? =>
+    javaScriptEnabled::bool? =>
+    mixedContentMode::list [ | `never | `always | `compatibility]? =>
+    thirdPartyCookiesEnabled::bool? =>
+    userAgent::string? =>
+    allowsInlineMediaPlayback::bool? =>
+    bounces::bool? =>
+    dataDetectorTypes::list [ | `phoneNumber | `link | `address | `calendarEvent | `none | `all]? =>
+    decelerationRate::list [ | `normal | `fast | `value float]? =>
+    onShouldStartLoadWithRequest::(webViewIOSLoadRequestEvent => bool)? =>
+    scrollEnabled::bool? =>
+    array ReasonReact.reactElement =>
+    ReasonReact.component ReasonReact.stateless ReasonReact.noRetainedProps;
+};
+
+module type Impl = {let webView: ReasonReact.reactClass;};
+
+module CreateComponent (Impl: Impl) :WebViewComponent => {
+  type webViewNavState;
+  external webViewNavState :
+    url::string? =>
+    title::string? =>
+    loading::bool? =>
+    canGoBack::bool? =>
+    canGoForward::bool? =>
+    unit =>
+    webViewNavState =
+    "" [@@bs.obj];
+  type webViewIOSLoadRequestEvent;
+  external webViewIOSLoadRequestEvent :
```

Comment:
I think this is the wrong way. `onShouldStartLoadWithRequest` accepts a cb (see above) which has an `Js.t {.…}` as input.
Now there are two ways to handle it.
1) Expose the Js.t object in the types, i.e. the signature I commented above
2) Expose a corresponding record type (say `iOSLoadRequestEvent`) and a function which takes as argument a function `iOSLoadRequestEvent  => unit`  and returns a function `Js.t {..} => unit`. See the discussion in https://github.com/BuckleTypes/bs-react-native/pull/37 for the some additional explanation in the case of SectionList.